### PR TITLE
Add `OpaquePointer` implementation

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This type is required for a basic `StaticString` implementation.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,14 +13,11 @@ jobs:
...
-            tag: 5.4.3-RELEASE
-
-          - branch: swift-5.5-release
-            tag: 5.5-RELEASE
+          - branch: swift-5.6-branch
+            tag: 5.6-DEVELOPMENT-SNAPSHOT-2022-02-11-a
 
           - branch: development
-            tag: DEVELOPMENT-SNAPSHOT-2021-09-18-a
+            tag: DEVELOPMENT-SNAPSHOT-2022-02-03-a
```
Comment (compnerd) [inline .github/workflows/build.yml:20]: These changes are unrelated to the addition of the new type.

File: Sources/Core/Int.swift [hunk 2]
```diff
@@ -8,22 +8,21 @@
 @frozen
 public struct Int {
   @usableFromInline
-  internal var _value: Builtin.Int32
+  internal var _value: Builtin.Word
```
Comment (compnerd) [inline Sources/Core/Int.swift:11]: This is unrelated to the addition of `OpaquePointer`.

File: Sources/Core/OpaquePointer.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  internal var _rawValue: Builtin.RawPointer
+
+  @usableFromInline @_transparent
+  internal init(_ v: Builtin.RawPointer) {
+    _rawValue = v
+  }
+
+  @_transparent
+  public init?(bitPattern: Int) {
+    guard bitPattern != 0 else { return nil }
```
Comment (compnerd) [inline Sources/Core/OpaquePointer.swift:17]: Is there a reason to use `guard` here rather than `if bitPattern == 0 { return nil }`?

File: Sources/Core/UInt.swift [hunk 4]
```diff
@@ -8,22 +8,21 @@
 @frozen
 public struct UInt {
   @usableFromInline
-  internal var _value: Builtin.Int32
+  internal var _value: Builtin.Word
```
Comment (compnerd) [inline Sources/Core/UInt.swift:11]: Unrelated to the addition of `OpaquePointer`.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: Sources/Core/OpaquePointer.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  @usableFromInline @_transparent
+  internal init(_ v: Builtin.RawPointer) {
+    _rawValue = v
+  }
+
+  @_transparent
+  public init?(bitPattern: Int) {
+    if bitPattern == 0 { return nil }
+    _rawValue = Builtin.inttoptr_Int32(bitPattern._value)
```
Comment (compnerd) [inline Sources/Core/OpaquePointer.swift:18]: This will truncate no?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -13,12 +13,6 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - branch: swift-5.4.3-release
-            tag: 5.4.3-RELEASE
-
-          - branch: swift-5.5-release
-            tag: 5.5-RELEASE
-
```
Comment (compnerd) [inline .github/workflows/build.yml:21]: IMO this is unrelated to implementing `OpaquePointer`.

File: Sources/Core/OpaquePointer.swift (hunk 5 above)
Comment (MaxDesiatov) [reply Sources/Core/OpaquePointer.swift:18]: It will, but I had to use `Int32` here, as you've requested `Word` changes to `UInt` and `Int` to be removed from this PR in your comment above.

File: .github/workflows/build.yml (hunk 6 above)
Comment (MaxDesiatov) [reply .github/workflows/build.yml:21]: This releases crash when attempting to build `OpaquePointer` code, as can be seen in #31. What would be the best way to make CI green on this PR without removing these releases from the build matrix?

Comment (compnerd) [reply .github/workflows/build.yml:21]: You can use `#if swift(>=5.6)`  around the implementation of `OpaquePointer` :)

File: Sources/Core/OpaquePointer.swift (hunk 5 above)
Comment (compnerd) [reply Sources/Core/OpaquePointer.swift:19]: I'm trying to get you to be more methodical about the changes.  Clearly there is a bug in `Int` and `UInt`, that should be fixed first.  Once that is fixed, adding an implementation here is simple.

File: Sources/Core/OpaquePointer.swift [hunk 7]
```diff
@@ -0,0 +1,50 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+#if swift(>=5.6)
```
Comment (compnerd) [inline Sources/Core/OpaquePointer.swift:5]: Newlines after this would've been nice, but eh.

**Review by compnerd — APPROVED** [review]

File: Sources/Core/OpaquePointer.swift [hunk 8]
```diff
@@ -0,0 +1,53 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+// This `OpaquePointer` implementation is known to crash 5.4 and 5.5 compiler releases on Windows.
```
Comment (compnerd) [inline Sources/Core/OpaquePointer.swift:5]: Thanks for the comment, that is a good idea to record!

**Review by compnerd — APPROVED** [review]

File: Sources/Core/OpaquePointer.swift (hunk 8 above)
Comment (MaxDesiatov) [reply Sources/Core/OpaquePointer.swift:5]: The formatter I use has done a stupid (in my opinion) thing by indenting everything within `#if` block. I've pushed one more time to fix that.
